wallet/src/de/schildbach/wallet/ui/dashpay/TransactionsViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package de.schildbach.wallet.ui.dashpay
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import de.schildbach.wallet.Constants
+import de.schildbach.wallet.WalletApplication
+import de.schildbach.wallet.data.DashPayProfile
+import de.schildbach.wallet.data.UsernameSortOrderBy
+import kotlinx.coroutines.launch
+import org.bitcoinj.core.Context
+import org.bitcoinj.core.Sha256Hash
+import org.bitcoinj.core.Transaction
+import org.bitcoinj.core.TransactionConfidence
+import org.bitcoinj.wallet.Wallet
+import java.util.*
+import kotlin.collections.HashMap
+
+class TransactionsViewModel(application: Application) : AndroidViewModel(application) {
+
+    enum class Direction {
+        RECEIVED, SENT
+    }
+
+    val direction = MutableLiveData<Direction?>()
+    val transactionsLiveData = MediatorLiveData<Pair<List<Transaction>,
+            Map<Sha256Hash, DashPayProfile>>>()
+    private val balanceLiveData = WalletBalanceLiveData()
+
+    init {
+        load()
+        transactionsLiveData.addSource(direction) {
+            load()
+        }
+        transactionsLiveData.addSource(balanceLiveData) {
+            load()
+        }
+    }
+
+    fun load(wallet: Wallet = WalletApplication.getInstance().wallet) {
```

Comment:
the `wallet` val should be moved inside the method body

Reply:
Fixed in https://github.com/dashevo/dash-wallet/pull/527/commits/55f7a80c91caee8315f35c12b42582a5b52f9f00